loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,388 @@
+    * [breaks](#the-breaks-field) — Collection of mods that this mod is incompatible with
+    * [repositories](#the-repositories-field) — Array of maven repositories
+    * [metadata](#the-metadata-field) — Extra information about this mod and/or its authors
+        * [name](#the-name-field) — Human-readable name of this mod
+        * [description](#the-description-field) — Human-readable description of this mod
+        * [contributors](#the-contributors-field) — Collection of contributors to this mod
+        * [contact](#the-contact-field) — Collection of contact information
+        * [license](#the-license-field) — One or more licenses this project is under
+        * [icon](#the-icon-field) — The icon or icons associated with this project
+* [mixin](#the-mixin-field) — Path(s) to mixin config file(s)
+* [access_widener](#the-access_widener-field) — Path(s) to accesswidener file(s)
+* [minecraft](#the-minecraft-field) - Minecraft related options
+    * [environment](#the-environment-field) — What game environments this mod applies to
+    
+## The `schema_version` field
+| Type   | Required |
+|--------|----------|
+| Number | True     |
+
+The quilt mod file schema version to be used for parsing this file. Currently, the only valid version is 1.
+
+## The `quilt_loader` field
+| Type   | Required |
+|--------|----------|
+| Object | True     |
+
+Information necessary for the mod loading process.
+
+### The `group_id` field
+| Type   | Required |
+|--------|----------|
+| String | True     |
+
+A unique identifier for the organization behind or developers of the mod. See Maven's [guide to naming conventions](https://maven.apache.org/guides/mini/guide-naming-conventions.html).
+
+### The `id` field
+| Type   | Required |
+|--------|----------|
+| String | True     |
+
+A unique identifier for the mod or library defined by this file, matching the `^[a-z][a-z0-9-_]{1,63}$` regular expression. Best practice is that mod ID's are in snake_case.
+
+### The `provides` field
+| Type         | Required |
+|--------------|----------|
+| Array/String | False     |
+
+Either a single identifier or an array of identifiers that this mod also contains. These could be old or alternative mod ids. They should follow [the convention established for mod ids](#the-mod_id-field).
+
+### The `version` field
+| Type   | Required |
+|--------|----------|
+| String | True     |
+
+Must conform to the [Semantic Versioning 2.0.0 specification](https://semver.org/). In a development environment, the value `${version}` can be used as a placeholder by quilt-gradle to be replaced on building the resulting JAR.
+
+### The `entrypoints` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+A collection of `key: value` pairs, where each key is the type of the entrypoints specified and each values is either a single entrypoint or an array of entrypoints. An entrypoint is an object with the following keys:
```

Comment:
Inner details of what an entrypoint entry can look like probably needs further documentation.